project-base/storefront/components/Pages/Order/ContactInformation/ContactInformationDeliveryAddress.tsx

```diff
@@ -72,71 +65,49 @@ export const ContactInformationDeliveryAddress: FC = () => {
                 updateContactInformation({ ...pickupPlace, country: selectedCountryOption });
             }
         }
-    }, [
-        pickupPlace,
-        differentDeliveryAddressValue,
-        countriesAsSelectOptions,
-        getValues,
-        setValue,
-        formMeta.fields.deliveryFirstName.name,
-        formMeta.fields.deliveryLastName.name,
-        formMeta.fields.deliveryCompanyName.name,
-        formMeta.fields.deliveryTelephone.name,
-        formMeta.fields.deliveryStreet.name,
-        formMeta.fields.deliveryCity.name,
-        formMeta.fields.deliveryPostcode.name,
-        formMeta.fields.deliveryCountry.name,
-        updateContactInformation,
-    ]);
+    }, [countriesAsSelectOptions]);
 
     useEffect(() => {
-        if (user) {
-            const deliveryAddress = user.deliveryAddresses.find((address) => address.uuid === deliveryAddressUuidValue);
-            const selectedCountryOption =
-                countriesAsSelectOptions.find((option) => option.value === deliveryAddress?.country) ??
-                countriesAsSelectOptions.find((option) => option.value === user.country.code);
+        if (user && deliveryAddressUuid) {
+            const deliveryAddress = user.deliveryAddresses.find((address) => address.uuid === deliveryAddressUuid)!;
+            const selectedCountryOption = countriesAsSelectOptions.find(
+                (option) => option.label === deliveryAddress.country,
```

Comment:
Wny are we searching base on the label? That seems incorrect

Reply:
What I mean is that you are comparing it with the label, whereas on lines 45-47 we have this code;

```ts
const selectedCountryOption = countriesAsSelectOptions.find(
    (option) => option.value === pickupPlace.country.
);
```

I think its safer to compare it with the value, as that is the ISO code. It is more consistent IMHO.

Reply:
Yes, because on the line which you've commented there is `deliveryAddress.country` which is returned from User API. Value which is being returned is label, not value.